server/server.go

```diff
@@ -1248,7 +1248,7 @@ func (s *Server) Start() {
 	// own system account if one is not present.
 	if opts.JetStream {
 		var cfg *JetStreamConfig
-		if opts.StoreDir != "" {
+		if opts.StoreDir != "" || opts.JetStreamMaxMemory != 0 || opts.JetStreamMaxStore != 0 {
```

Comment:
There's some other problems here:

https://github.com/nats-io/nats-server/blob/9ac7dfd9abab32eb49ddaddac825ca78a1c4a78a/server/jetstream.go#L102-L109

Here unless both memory and file is set it will use the dynamic config so `jetstream: {max_mem_store: 2GB }` yields

```
----------- JETSTREAM (Beta) -----------
  Max Memory:      12.00 GB
  Max Storage:     1.00 TB
```

Might be easier to set it to dynamic config upgfront and then change the settings from config file over onto dynamic, so you get dynamic settings for unset parts.  Also be less error prone in future, now munging cfg is spread over multiple functions
```

Reply:
ok I just fixed this in another commit.